extension/js/content_scripts/webmail/gmail/gmail-element-replacer.ts

```diff
@@ -291,10 +291,13 @@ export class GmailElementReplacer extends WebmailElementReplacer {
   };
 
   private replaceActionsMenu = () => {
-    if ($('.action_menu_message_button').length <= 0) {
-      this.addMenuButton('reply', '#r');
-      this.addMenuButton('forward', '#r3');
-    }
+    $(document).on('click', 'div.aHU.hx', event => {
+      const $actionsBtn = $(event.currentTarget).find(this.sel.msgActionsBtn);
+      if ($actionsBtn.length && !$('.action_menu_message_button').length) {
+        this.addMenuButton('reply', '#r');
+        this.addMenuButton('forward', '#r3');
+      }
+    });
   };
```

Comment:
with current implementation `replaceActionsMenu` will be executed each second, adding more and more `$(document).on('click')` handlers which can lead to performance issues.

it'll be more correct to add this handler in [GmailWebMailStartup](https://github.com/FlowCrypt/flowcrypt-browser/blob/27b8b376642d92f4efb1491aeb1412c9ef7daa9e/extension/js/content_scripts/webmail/gmail/gmail-webmail-startup.ts#L43), as it'll be executed only once and handle all future clicks:

```ts
private addSecureActionsToMessageMenu = () => {
  $('body').on('click', 'div.aHU.hx', event => {
    this.replacer.addSecureActionsToMessageMenu(event.currentTarget)
  });
}
```

Reply:
Thank you @sosnovsky for your constant guidance. I see what you mean by now, its the heartbeat like injection secure compose which may lead to performance issue. I have done the necessary changes and the result is good.